mlflow/server/js/src/experiment-tracking/components/CompareRunView.js

```diff
@@ -258,54 +377,71 @@ export class CompareRunView extends Component {
   // eslint-disable-next-line no-unused-vars
   renderDataRows(
     list,
-    highlightChanges = false,
+    colWidth,
+    onlyShowDiff,
+    highlightDiff = false,
     headerMap = (key, data) => key,
     formatter = (value) => value,
   ) {
-    const keys = CompareRunUtil.getKeys(list);
+    let keys = CompareRunUtil.getKeys(list);
     const data = {};
     keys.forEach((k) => (data[k] = []));
     list.forEach((records, i) => {
       keys.forEach((k) => data[k].push(undefined));
       records.forEach((r) => (data[r.key][i] = r.value));
     });
 
+    const colWidthStyle = {
+      width: `${colWidth}px`,
+      minWidth: `${colWidth}px`,
+      maxWidth: `${colWidth}px`,
+    };
+
+    function isAllEqual(k) {
+      return data[k].every((x) => x === data[k][0]);
+    }
+
+    if (onlyShowDiff) {
+      keys = keys.filter((k) => !isAllEqual(k));
+    }
     return keys.map((k) => {
-      let row_class = undefined;
-      if (highlightChanges) {
-        const all_equal = data[k].every((x) => x === data[k][0]);
-        if (!all_equal) {
-          row_class = 'row-changed';
-        }
+      const allEqual = isAllEqual(k);
```

Comment:
can we avoid calling `isAllEqual` twice?

Reply:
yes. we can simply return `null` in the forEach function for hidden rows.